packages/cerebral-provider-firebase/src/createTask.js

```diff
@@ -5,20 +5,25 @@ export default function createTask (options) {
     return new Promise((resolve, reject) => {
       const tasksPath = options.queuePath ? options.queuePath + '/tasks' : 'queue/tasks'
       const ref = firebase.database().ref(tasksPath)
-      const taskKey = ref.push(Object.assign({
-        _state: options.specPrefix ? `${options.specPrefix}_${name}` : name
-      }, payload))
 
-      const taskRef = firebase.database().ref(`${tasksPath}/${taskKey.key}`)
-      taskRef.on('value', data => {
-        if (!data.val()) {
-          taskRef.off()
-          resolve()
-        }
-        if (data.val() && data.val()._error_details) {
-          reject(data.val()._error_details)
-        }
-      })
+      return firebase.auth().currentUser.getToken()
+        .then((_token) => {
+          const taskKey = ref.push(Object.assign({
+            _state: options.specPrefix ? `${options.specPrefix}_${name}` : name,
+            _token
+          }, payload))
+
+          const taskRef = firebase.database().ref(`${tasksPath}/${taskKey.key}`)
+          taskRef.on('value', data => {
+            if (!data.val()) {
```

Comment:
maybe add:

```js
const value = data.val()
```

Not sure about the cost of "data.val()"... But maybe it's very small.